File: src/main/java/autotransform/input/CachedFile.java
package autotransform.input;

import java.io.File;
import java.io.IOException;
import java.util.Scanner;

public class CachedFile {
    private String path;
    private String content;

    public CachedFile(String path) {
        this.path = path;
    }

    public String getContent() throws IOException {
        if (this.content == null) {
            File file = new File(this.path);
            Scanner scanner = new Scanner(file);
            StringBuilder builder = new StringBuilder();
            if (scanner.hasNext()) {
                builder.append(scanner.nextLine());
            }
            while (scanner.hasNext()) {
                builder.append("\n");
                builder.append(scanner.nextLine());
            }
            scanner.close();
            this.content = builder.toString();
        }

        return this.content;
    }

    public String getPath() {
        return this.path;
    }
}

File: src/common/cachedfile.py
from typing import Optional

class CachedFile:
    def __init__(self, path: str):
        self.path = path
        self.content: Optional[str] = None
    
    def get_content(self) -> str:
        if self.content is None:
            f = open(self.path, 'r')
            self.content = f.read()
            f.close()
        return self.content